### DISCUSSION TITLE
Prevent Typer from interpreting list of parameters as options when starting with "--"

### DISCUSSION THREAD

**Discussion by offerrall** [fastapi/typer#550] [Questions]
I have a function that accepts a parameter and a list of parameters, however sometimes the list of parameters includes arguments that start with "--" which causes Typer to interpret them as options. Here's the code for reference:
```

@app.command("new")
def new(alias: str, command: list[str]):
    '''Create a new alias, EX: pyAlias new ls dir /b'''
    command = " ".join(command)
    create_alias(alias, command, config)
    print(f"Alias created: {alias} -> {command}")

```

Is there a way to prevent Typer from checking these parameters as options and instead treat them as part of the list of parameters? Any help would be appreciated.

**Comment by PurpleBooth** [comment]
I have a similar problem with negative numbers that are args being interpreted as option

**Reply by NikosAlexandris** [reply]
In the relevant [Click](https://github.com/pallets/click) [issue 1624](https://github.com/pallets/click/issues/1624), see [comment 656903382](https://github.com/pallets/click/issues/1624#issuecomment-656903382) which points to: https://click.palletsprojects.com/en/7.x/arguments/?highlight=ignore_unknown_options#option-like-arguments. Not sure yet how to plug and play this in Typer.

**Reply by NikosAlexandris** [reply]
The answer is here : https://typer.tiangolo.com/tutorial/commands/context/, see the example at https://typer.tiangolo.com/tutorial/commands/context/#configuring-the-context.
